cachito/workers/pkg_managers/general_js.py

```diff
@@ -0,0 +1,273 @@
+    # //nexus.domain.local/repository/cachito-js-1/:_authToken=NpmToken.0fe010d3-8225-3f03-b4c4-662db1573f9f
+    npm_rc = textwrap.dedent(
+        f"""\
+        registry={proxy_repo_url}
+        //{proxy_repo_url[len(proxy_transport):]}:_authToken={token}
+        fetch-retries=5
+        fetch-retry-factor=2
+        """
+    )
+    nexus_ca_cert = nexus.get_ca_cert()
+    if nexus_ca_cert:
+        npm_rc = textwrap.dedent(
+            f"""\
+            {npm_rc}
+            ca="{nexus_ca_cert}"
+            strict-ssl=true
+            """
+        )
+
+    return npm_rc
+
+
+def generate_npmrc_file(npm_rc_path, request_id, token):
+    """
+    Generate an .npmrc file at the input location with the registry and authentication configured.
+
+    :param str npm_rc_path: the path to create the .npmrc file
+    :param int request_id: the ID of the request to determine the npm registry URL
+    :param str token: the npm token to use for the registry
+    """
+    log.debug("Generating the .npmrc file at %s", npm_rc_path)
+    with open(npm_rc_path, "w") as f:
+        f.write(generate_npmrc(request_id, token))
+
+
+def get_js_proxy_repo_name(request_id):
+    """
+    Get the name of npm proxy repository for the request.
+
+    :param int request_id: the ID of the request this repository is for
+    :return: the name of npm proxy repository for the request
+    :rtype: str
+    """
+    return f"cachito-js-{request_id}"
+
+
+def get_js_proxy_repo_url(request_id):
+    """
+    Get the URL for the Nexus npm proxy repository for the request.
+
+    :param int request_id: the ID of the request this repository is for
+    :return: the URL for the Nexus npm proxy repository for the request
+    :rtype: str
+    """
+    config = get_worker_config()
+    repo_name = get_js_proxy_repo_name(request_id)
+    return f"{config.cachito_nexus_url.rstrip('/')}/repository/{repo_name}/"
+
+
+def get_js_proxy_username(request_id):
+    """
+    Get the username that has read access on the npm proxy repository for the request.
+
+    :param int request_id: the ID of the request this repository is for
+    :return: the username
+    :rtype: str
+    """
+    return f"cachito-js-{request_id}"
+
+
+def get_npm_privileged_token(request_id):
+    """
+    Get an npm token for the privileged Nexus user for the npm proxy repository for the request.
+
+    :param int request_id: the request ID of the Nexus npm proxy repository to get the token for
+    :return: the npm token to be set in the ``.npmrc`` file
+    :rtype: str
+    :raise CachitoError: if the token cannot be retrieved
+    """
+    conf = get_worker_config()
+    return get_npm_token(request_id, conf.cachito_nexus_username, conf.cachito_nexus_password)
+
+
+def get_npm_token(request_id, username, password):
+    """
+    Get an npm token for the Nexus npm proxy repository for the request.
+
+    This is an alternative to calling `npm login`, which doesn't have the ability to accept the
```

Comment:
The token does not expire from what I can tell.

I'll look into trying this approach out instead to avoid this function:
https://help.sonatype.com/repomanager3/formats/npm-registry#npmRegistry-AuthenticationUsingBasicAuth